### PR TITLE
Weston v4 adaptations

### PR DESCRIPTION
**Pull request by eucan**
Update wayland-ivi-extension to use weston 4.0

### CODE REVIEW

File: weston-ivi-shell/src/ivi-controller.c [hunk 1]
```diff
@@ -1340,6 +1340,7 @@ controller_screen_screenshot(struct wl_client *client,
     l->frame_listener.notify = controller_screenshot_notify;
     wl_signal_add(&iviscrn->output->frame_signal, &l->frame_listener);
     iviscrn->output->disable_planes++;
+    weston_output_damage(iviscrn->output);
     weston_output_schedule_repaint(iviscrn->output);
```
[the remark this answers is not in the document]
Comment (eucan) [reply weston-ivi-shell/src/ivi-controller.c]: you are right. good catch. I will fix this

File: weston-ivi-shell/src/ivi-controller.c [hunk 2]
```diff
@@ -1340,8 +1340,7 @@ controller_screen_screenshot(struct wl_client *client,
     l->frame_listener.notify = controller_screenshot_notify;
     wl_signal_add(&iviscrn->output->frame_signal, &l->frame_listener);
     iviscrn->output->disable_planes++;
-    weston_output_schedule_repaint(iviscrn->output);
-    return;
+    weston_output_damage(iviscrn->output);
```
Comment (efriedrich) [inline weston-ivi-shell/src/ivi-controller.c]: it seams not to be related to the weston 4.0 update, we should create a separate PR and merge it before

**Comment by eucan** [comment]
@efriedrich I got nofitication email about your email. But I don't see your review findings in PR.
Anyway you said that:
"the patch 1a011da should be a separate PR and we should merge it before the current one,
patches
a25c251 and
fd860d2 are also not related to weston 4.0 right but since there is a dependency to the weston.ini so kind of public interface it should be a separate PR and we shold merge it after the weston 4.0 PR
"

First of all, a25c251 and fd860d2 are an integral part of this PR. Weston IVI Shell would not load ivi-controller without these patches. If we merge other patches and not these two patches, wayland-ivi-extension would be in an unusable state.

In weston 4.0, they changed behavior of weston_output_schedule_repaint(). Previously, it triggered repaint of output regardless of damage. In weston 4.0, output is not repainted if there are no damage on it. Screenshot unit-tests are failing without this patch. Therefore, the patch 1a011da should be either merged before this PR or it should be part of this PR. I don't see any benefit to put the patch in another PR. If you insist, I can do it. No problem

**Comment by efriedrich** [comment]
@eucan 
>@efriedrich I got nofitication email about your email. But I don't see your review findings in 

this is strange seam to be removed I deleted some comment inside the patch itself hm...
but O.K. in general my assumption about the independence from weston 4.0 is wrong the patches are very well dependent and necessary for weston 4.0 update:
* 1a011da  is an changed behavior of the api and
* for a25c251 and fd860d2 there should be a change in weston otherwise I'm wondering how it is working now?

if the two point above are correct no further objections from my side we can merge

**Comment by eucan** [comment]
@efriedrich 
weston changes are already part of weston 4.0: 0c1bbb9 and ce9bc35.

Can I put your reviewed-by or acked-by ?

**Pull request merged**